frontend/src/metabase/components/PaginationControls/PaginationControls.tsx

```diff
@@ -0,0 +1,72 @@
+import { t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import { Group, Text } from "metabase/ui";
+
+const isLastPage = (pageIndex: number, pageSize: number, total: number) =>
+  pageIndex === Math.ceil(total / pageSize) - 1;
+
+export type PaginationControlsProps = {
+  page: number;
+  pageSize: number;
+  itemsLength: number;
+  total?: number;
+  showTotal?: boolean;
+  onNextPage?: (() => void) | null;
+  onPreviousPage?: (() => void) | null;
+};
+
+export const PaginationControls = ({
+  page,
+  pageSize,
+  itemsLength,
+  total,
+  showTotal = false,
+  onNextPage,
+  onPreviousPage,
+}: PaginationControlsProps) => {
+  const isSinglePage = total !== undefined && total <= pageSize;
+
+  if (isSinglePage) {
+    return null;
+  }
+
+  const isPreviousDisabled = page === 0;
+  const isNextDisabled =
+    total != null ? isLastPage(page, pageSize, total) : !onNextPage;
+
+  return (
+    <Group align="center" fw="bold" aria-label="pagination">
+      <Text span mr="sm">
+        {page * pageSize + 1} - {page * pageSize + itemsLength}
+        {showTotal && (
+          <>
+            <Text span c="text-light">
+              &nbsp;{t`of`}&nbsp;
```

Comment:
```suggestion
              &nbsp;{c("Appears in phrases like '1-10 of 100', referring to a page of results").t`of`}&nbsp;
```

Reply:
I admit this is one of my hobby-horse issues, but it's so hard to translate little words like this out of context, and pretty easy to provide that context with `c`, so (tho you're just porting) it might be easy to throw the translators a lifeline here

Reply:
As you may know `c` adds a bit of context in POEditor. That looks like this:

![image](https://github.com/metabase/metabase/assets/130925/15407d35-369c-47ca-94ea-a011c63adcdc)

When you hover it shows the whole string

![image](https://github.com/metabase/metabase/assets/130925/c18acb66-261e-47d6-8865-82eac19af39b)

Reply:
I knew something like this existed but didn't know about the syntax - was _very_ jarring at first but I like it :smile:  I'll add it in